### PR TITLE
ref: allow hosted js sdk bundles

### PR DESCRIPTION
**Pull request by aldy505**
Since this is hanging on my branch, I guess I'll make it out here. A part of making the JS SDK loader script available on the self-hosted sentry. Refer to https://github.com/getsentry/sentry/issues/22715.

Since not everyone is happy on hosting the JS SDK themself, therefore I made it into a feature flag. Set `SETUP_JS_SDK_ASSETS=1` on the `.env` file to enable the script. Also set `SETUP_JS_SDK_KEEP_OLD_ASSETS=1` to never delete (or basically cleanup) previous JS SDK version.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -88,6 +88,10 @@ http {
 		location ^~ /api/0/relays/ {
 			proxy_pass http://relay;
 		}
+		location ^~ /js-sdk/ {
+			autoindex on;
+			root  /var/www/js-sdk;
+		}
```
Comment (aldy505) [inline nginx/nginx.conf:94]:
```suggestion
		}
```

**Comment by BYK** [comment]
Wow, this looks quite nice! I'd make it opt-out rather than opt-in as it the assets shouldn't take too much space and this is a great quality of life improvement.

I still don't get the reasoning behind using an external volume compared to a project-specific volume tho. Can you expand on that a bit?

**Comment by aldy505** [comment]
> Wow, this looks quite nice! I'd make it opt-out rather than opt-in as it the assets shouldn't take too much space and this is a great quality of life improvement.
> 
> 
> 
> I still don't get the reasoning behind using an external volume compared to a project-specific volume tho. Can you expand on that a bit?

@BYK I made it quite a while ago so I don't really remember. Probably because the nginx container has web container as the dependency, so I think we shouldn't start the nginx container at all. 

If there is something like "docker compose volume create", that'd be nice.

**Comment by BYK** [comment]
>@BYK I made it quite a while ago so I don't really remember. Probably because the nginx container has web container as the dependency, so I think we shouldn't start the nginx container at all.

I mean you don't need to spin up `nginx` (and its dependencies) to be able to use the volume? We can bind that to a different service or just use `--no-deps` to not start any other dependencies. You may add a pseudo service with a base Debian image that defines all volumes on it and use that as the "install helper". We can add that to a profile such as `installer` so it is never included in regular setups.

WDYT?

**Comment by aldy505** [comment]
> > @BYK I made it quite a while ago so I don't really remember. Probably because the nginx container has web container as the dependency, so I think we shouldn't start the nginx container at all.
> 
> I mean you don't need to spin up `nginx` (and its dependencies) to be able to use the volume? We can bind that to a different service or just use `--no-deps` to not start any other dependencies. You may add a pseudo service with a base Debian image that defines all volumes on it and use that as the "install helper". We can add that to a profile such as `installer` so it is never included in regular setups.
> 
> WDYT?

In the end, I used `--no-deps`.

But I'd still prefer making this opt-in instead. I don't know about other people though.

**Comment by aldy505** [comment]
Finally.

File: docker-compose.yml [hunk 2]
```diff
@@ -529,6 +530,10 @@ volumes:
     external: true
   sentry-symbolicator:
     external: true
+  # This volume stores JS SDK assets, this is created as an external volume
+  # to minimize hiccups created by the nginx container and its' dependencies.
+  # The data inside this volume should be cleaned periodically on upgrades.
```
Comment (BYK) [inline docker-compose.yml:535]: Just helping you out :)

```suggestion
  # This volume stores JS SDK assets and the data inside this volume should
  # be cleaned periodically on upgrades.
```

File: install/setup-js-sdk-assets.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# This will only run if the SETUP_JS_SDK_ASSETS environment variable is set to 1
+if [[ "${SETUP_JS_SDK_ASSETS:-}" == "1" ]]; then
```
Comment (hubertdeng123) [inline install/setup-js-sdk-assets.sh:2]: IMO if we're going to include this as part of the install script moving forwards, I'd like to see it in the `.env` file just to surface this as something people can set up. We can leave it as default for opt-in

**Review by hubertdeng123 — COMMENTED** [review]
Looks pretty good, just one minor comment. This is awesome work!

File: install/setup-js-sdk-assets.sh (hunk 3 above)
Comment (aldy505) [reply install/setup-js-sdk-assets.sh:3]: Oh not you too :(

Comment (aldy505) [reply install/setup-js-sdk-assets.sh:3]: Oh sorry I read that wrong. Lol.

Comment (aldy505) [reply install/setup-js-sdk-assets.sh:3]: Okay done

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**